salt/utils/aws.py

```diff
@@ -475,16 +477,24 @@ def query(params=None, setname=None, requesturl=None, location=None,
 
             # check to see if we should retry the query
             err_code = data.get('Errors', {}).get('Error', {}).get('Code', '')
-            if attempts > 0 and err_code and err_code in AWS_RETRY_CODES:
-                attempts -= 1
+            if attempts < MAX_RETRIES and err_code and err_code in AWS_RETRY_CODES:
```

Comment:
Couldn't we can skip the check for `attempts < MAX_RETRIES` here, since this is the condition for the while loop?